### PR TITLE
Euler angles

### PR DESCRIPTION
**Pull request by djowatts**
Giving the ability to represent a quaternion as euler angles

### CODE REVIEW

File: src/Spatial/Euclidean/EulerAngles.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using MathNet.Spatial.Units;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// A means of representing spatial orientation of any reference frame. 
+    /// More information can be found https://en.wikipedia.org/wiki/Euler_angles
+    /// </summary>
+    public class EulerAngles
+    {
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/EulerAngles.cs]: Not a struct?

File: src/Spatial/Euclidean/Quaternion.cs [hunk 2]
```diff
@@ -155,6 +156,24 @@ public bool IsRotation
...
+        /// <returns></returns>
+        public EulerAngles ToEulerAngles()
+        {
+            return new EulerAngles(
+                new Angle(Math.Atan2(2*(_w*_x + _y*_z), ((_w*_w) + (_z*_z) - (_x*_x) - (_y*_y))),
+                    AngleUnit.Radians),
+                new Angle(Math.Asin(2*(_w*_y - _x*_z)), AngleUnit.Radians),
+                new Angle(Math.Atan2(2*(_w*_z + _x*_y), ((_w*_w) + (_x*_x) - (_y*_y) - (_z*_z))),
+                    AngleUnit.Radians));
+        }
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/Quaternion.cs]: Unit tests are nice :)

**Comment by JohanLarsson** [comment]
Maybe squash the commits?

**Comment by djowatts** [comment]
Popped in some basic tests and switched to a struct. Cheers for the feedback. Also, this is the first time I've done commit squashing (my word that was fun with already pushed commits). 

Thanks for the feedback :+1:

**Comment by JohanLarsson** [comment]
Thank you sir.

**Pull request merged**